polyphemus/lib/etls/ipi/ipi_etls.rb

```diff
@@ -115,10 +113,13 @@ def process(cursor, folders)
 
           matched_folders = folders.map do |folder|
             record_name = @linker.record_name_by_path(
-              folder.folder_path + '/',
```

Comment:
the record name regex works well for folders --- since file names themselves can also often match the ipi record regex, but are not identifiers non the less.

To make this system work better, record_name_by_path accepts either folders or files and changes its logic to better match based on the target.  See other usages in this PR.